src/app/shared/services/share/share.service.ts

```diff
@@ -30,90 +37,140 @@ export class ShareService extends SyncServiceBase {
 
   private registerTemplateActionHandlers() {
     this.templateActionRegistry.register({
-      share: async ({ args }) => {
-        const [actionId, ...shareArgs] = args;
-        const childActions = {
-          file: async () => await this.shareFile(shareArgs[0]),
-          text: () => this.share({ text: shareArgs[0] }),
-          url: () => this.share({ url: shareArgs[0] }),
-        };
-        // To support deprecated "share" action (previously used to share text only),
-        // assume text is being shared if first arg is not an actionId
-        if (!(actionId in childActions)) {
-          return await this.share({ text: args[0] });
+      share: async (action) => {
+        const { args, params } = action as { args: string[]; params: IShareActionParams };
+
+        if (params) {
+          await this.handleShare(params);
+        } else if (args) {
+          console.warn("[SHARE] Deprecated action syntax. Use `share | data_type: data` instead.");
+          await this.handleLegacyShare(args);
+        } else {
+          return console.error("[SHARE] No params provided to `share` action");
         }
-        return childActions[actionId]();
       },
     });
   }
 
-  async share(options: ShareOptions) {
-    const { value: canShare } = await Share.canShare();
-    if (canShare) {
-      try {
-        const { activityType } = await Share.share(options);
-        console.log("[SHARE] Content shared to", activityType);
-      } catch (error) {
-        this.handleShareError(error);
-      }
-    } else console.error(SHARE_NOT_SUPPORTED_ON_PLATFORM_ERROR_MESSAGE);
+  private async handleShare(options: IShareActionParams) {
+    // Rename `dialog_title` to `dialogTitle`
+    let parsedOptions = {
+      dialogTitle: options.dialog_title,
+      ...(({ dialog_title, ...rest }) => rest)(options),
```

Comment:
nit(blocking): there's a log happening in this inline statement. Could you extract first for better clarity? E.g.
```ts
const {dialog_title,...shareOptions} = options
const parsedOptions = {dialogTitle: dialog_title,...shareOptions}
```